subsys/net/ip/net_if.c

```diff
@@ -555,6 +561,90 @@ static void dad_timeout(struct k_work *work)
 		 */
 		net_ipv6_nbr_rm(iface, &ifaddr->address.in6_addr);
 	}
+
+	ifaddr->dad.timer_timeout = 0;
+
+	sys_slist_find_and_remove(&active_dad_timers, &ifaddr->dad.node);
+}
+
+static bool dad_check_timeout(s64_t start, u32_t time, s64_t timeout)
+{
+	start += time;
+	start = abs(start);
+
+	if (start > timeout) {
+		return false;
+	}
+
+	return true;
+}
+
+static bool dad_timedout(struct net_if_addr *ifaddr, s64_t timeout)
+{
+	return dad_check_timeout(ifaddr->dad.timer_start,
+				 ifaddr->dad.timer_timeout,
+				 timeout);
+}
+
+static u32_t dad_manage_timeouts(struct net_if_addr *ifaddr, s64_t timeout)
+{
+	s32_t next_timeout;
+
+	if (dad_timedout(ifaddr, timeout)) {
+		dad_expired(ifaddr);
+		return UINT32_MAX;
+	}
+
+	next_timeout = timeout - (ifaddr->dad.timer_start +
+				  ifaddr->dad.timer_timeout);
+	return abs(next_timeout);
+}
+
+static void dad_timeout(struct k_work *work)
+{
+	u32_t timeout_update = UINT32_MAX - 1;
```

Comment:
Can (and should, to avoid confusion) be UINT32_MAX, similar or comment below.

Reply:
Yes, that can be done